File: sample-projects/go/domain/service/creature_service_test.go
package service_test

import (
	"testing"

	entity "de.sots/cellarsandcentaurs/adapter/persistence"
	"de.sots/cellarsandcentaurs/domain/model"
	"de.sots/cellarsandcentaurs/domain/service"
)

func Test_should_save_creature_to_the_stable(t *testing.T) {
	creatures := entity.NewPersistedCreatures(entity.NewCreatureRepository())
	creatureService := service.NewCreatureService(creatures)
	creature := model.NewCreature(model.NewCreatureId())
	walking_speed := model.NewSpeed(30)
	creature.SetSpeeds(map[model.SpeedType]model.Speed{model.Walking: walking_speed})

	err := creatureService.Save(creature)

	if err != nil {
		t.Fatalf("expected the creature to be saved, got %v", err)
	}
	found, _ := creatureService.Find(creature.Id())
	if found.Id() != creature.Id() {
		t.Errorf("expected creature %s in the stable", creature.Id())
	}
}

File: sample-projects/php/tests/Domain/Service/CreatureServiceTest.php
<?php

declare(strict_types=1);

namespace De\Sots\CellarsAndCentaurs\Tests\Domain\Service;

use De\Sots\CellarsAndCentaurs\Domain\Model\Creature;
use De\Sots\CellarsAndCentaurs\Domain\Model\CreatureId;
use De\Sots\CellarsAndCentaurs\Domain\Model\Speed;
use De\Sots\CellarsAndCentaurs\Domain\Model\SpeedType;
use De\Sots\CellarsAndCentaurs\Domain\Service\Creatures;
use De\Sots\CellarsAndCentaurs\Domain\Service\CreatureService;
use PHPUnit\Framework\TestCase;
use Psr\Log\NullLogger;

final class CreatureServiceTest extends TestCase
{
    public function test_should_save_creature_to_the_stable(): void
    {
        $creatures = $this->createMock(Creatures::class);
        $creatures->expects($this->once())->method('save');
        $service = new CreatureService($creatures, new NullLogger());
        $creature = new Creature(new CreatureId('centaur-1'));
        $walking_speed = new Speed(40);
        $creature->setSpeed(SpeedType::WALKING, $walking_speed);

        $service->save($creature);
    }
}
